lib/src/models/comment_impl.dart

```diff
@@ -60,7 +70,9 @@ class MoreComments extends RedditBase {
       : _children = data['children'],
         _count = data['count'],
         _parentId = data['parent_id'],
-        super.loadData(reddit, data);
+        super(reddit) {
+    setData(this, data);
```

Comment:
Since setData is a void function that is outside the scope of the super class (and abstract super class of that class), how are we able to call it here? Also what's with opening another scope after super, i.e the signature `SUPER LPAREN PARAM RPAREN LCURLY FUNC RCURLY` or in more concrete terms:
```
( 
super(reddit) {
     setData(this, data); 
}

Reply:
`setData` is defined as a top-level method in `base_impl.dart`, so we can call that from here. Let me know if there's still something that isn't super clear (there's some interesting stuff going on here with libraries and scoping here that might not be obvious).

Reply:
Do you have a name for the interesting behaviour, or at least some sort of string I could query google to learn more about it/docs reference?

Thanks!